### PR TITLE
Fix compiling python-bindings

### PR DESCRIPTION
**Pull request by masterleinad**
This should fix the failing build https://cdash.43-1.org/viewBuildError.php?buildid=2914.

### CODE REVIEW

File: contrib/python-bindings/include/cell_accessor_wrapper.h [hunk 1]
```diff
@@ -193,22 +193,22 @@ namespace python
   };
 
 
-  template <int dim, int spacedim>
+  template <int dim_, int spacedim_>
```
Comment (Rombur) [inline contrib/python-bindings/include/cell_accessor_wrapper.h]: Can you do the same change at line 175

File: contrib/python-bindings/include/cell_accessor_wrapper.h [hunk 2]
```diff
@@ -193,22 +193,22 @@ namespace python
...
   const CellAccessorWrapper
   CellAccessorWrapper::construct_neighbor_wrapper(const int i) const
   {
-    CellAccessor<dim, spacedim> *cell =
-      static_cast<CellAccessor<dim, spacedim> *>(cell_accessor);
+    Assert(dim_ == dim && spacedim_ == spacedim, ExcInternalError());
```
Comment (bangerth) [inline contrib/python-bindings/include/cell_accessor_wrapper.h]: You could make this a `static_assert`.

**Comment by masterleinad** [comment]
@Rombur Here you go!

File: contrib/python-bindings/include/cell_accessor_wrapper.h (hunk 2 above)
Comment (masterleinad) [reply contrib/python-bindings/include/cell_accessor_wrapper.h]: No, `dim` and `spacedim` are member variables. Apart from shadowing, this is the source of the compile errors in the first place.

Comment (bangerth) [reply contrib/python-bindings/include/cell_accessor_wrapper.h]: Oh yes, of course!

**Pull request merged**